server/src/controllers/meals.js

```diff
@@ -96,7 +98,9 @@ export const getMeals = async (req, res, next) => {
 
     console.log(query)
```

Comment:
ამას ამოვშლი ქუერის ვუყურებდი. გოგა მგონი შეცვლა მომიწევს ფილტრის მეთოდის.
მე ხომ parsedDate & parsedTime ობიექტების მიხედვით ვფილტრავ ,მაგაშია პრობლემა მგონი

![queryProblem](https://user-images.githubusercontent.com/68153140/150786251-9d14c450-d959-4a1c-b2ae-e59b8a577cc9.png)
 
ამ ქუერიში,  gte:4 და lte:1  არ აბრუნებს არაფერს,რადგან  4ზე მეტს ან ტოლს და 1ზე ნაკლებს ან ტოლს ვერ დააბრუნებს,და date პროპერთი რომ მაგაზე ხომ არა გადავაწყო ფილტრი.

ან რავიცი  მე ვაკეთებ რაღაცას ნიტოს

Reply:
გამომრიცხავი query არის, 4-ზე მეტი და 1-ზე ნაკლები ხომ ვერ იქნება.

ეს თუ მთლიანი ქვერია — იუზერის შეცდომაა და არა შენი. შეიძლება მან 4 ივნისი აირჩია პირველ თარიღში და მეორეში 1 ივლისი, შენ არაფერ შუაში ხარ.

ჩვენ შეგვიძლია კლიენტის მხარეს ისე ავაწყოთ კომპონენტები, რომ დავეხმაროთ ქვერის სწორად შედგენაში.

მაგალითად ინფუთები დავსვათ picker-ს მაგივრად.

მოდი ლოგიკაზე ბევრ დროს ნუღარ დავკარგავთ.

სხვა მიმართულებით წავიდეთ და როგორც შენ აკეთებდი ისე გავაკეთოთ.

from — to თარიღი. პირდაპირ თარიღები შევადაროთ და ეგ იქნება.

თარიღების შედარება ყოველთვის საკამათოა, როგორ უნდა შეინახო date მონაცემთა ბაზაში? ხანდახან timestamp-ს ინახავენ. 

რადგან მონგო document database არის, მგონი პირდაპირ date-ც შეგვიძლია შევინახოთ.

query-ში კი გეხნება მხოლოდ dateFrom და dateTo, ISOString-ები.

ჩვენ timestamp შევინახოთ (elapsed seconds since unix epoch).

Check:
* [Date.prototype.getTime()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime)
* [Info about dates in mongodb](https://www.compose.com/articles/understanding-dates-in-compose-mongodb)

Reply:
--> ჩვენ timestamp შევინახოთ (elapsed seconds since unix epoch).
1)სქემას რა ვუქნა . ესე გამოიყურება,
![schema](https://user-images.githubusercontent.com/68153140/151657047-7168dc05-83ff-46e5-b41a-1f2e24813866.png)


 timestamp-ი ჩამინაცვლებს dates, თუ დავტოვო ყველაფერი როგორც არის და დავამატო timestamp და მაგის მიხედვით გავფილტრო. 
(რაღაც ძალიან ბევრი date oriented პროპერთები დამიგროვდა )

2)-->ასევე, შექმნის დროს, picker რა თარიღსაც მოგცემს, მაგის getTime გაგზავნე და ეგ შეინახე სერვერზე.
როდესაც create meal ფუნქციონალს დავამატებ ფრონტში მაგ დროს ვქნა ეგ ანუ.

Reply:
ეს ყველაფერი უნდა წაშალო.

მხოლოდ date დაგრჩება, რომელიც იქნება timestamp. timestamp-ს ცალკე property-დ აღწერა არ ხდება. timestamp value არის თვითონ და არის Number

```js
date: {
   type: Number,
   required: true
}
```

time-ც აღარ გჭირდება new Date(date)-ს რომ იზამ ზუსტად იმ საათებსაც მოგცემს რაც აირჩიე